src/main/java/com/google/cloud/teleport/newrelic/utils/ConfigHelper.java

```diff
@@ -15,9 +21,18 @@ private ConfigHelper() {
      * @param <T> The type of the value being read
      * @return The value included in the provided ValueProvider, if it's available and non-null, otherwise the default value.
      */
-    public static <T> T valueOrDefault(ValueProvider<T> value, T defaultValue ) {
+    public static <T> ValueProvider<T> valueOrDefault(ValueProvider<T> value, T defaultValue ) {
         return (value != null && value.isAccessible()) && value.get() != null
-                ? value.get()
-                : defaultValue;
+                ? StaticValueProvider.of(value.get())
+                : StaticValueProvider.of(defaultValue);
+    }
+
+    public static void logConfiguration(final NewRelicConfig newRelicConfig) {
+        LOG.info("Logs API url set to: {}", newRelicConfig.getLogsApiUrl());
+        LOG.info("Batch count set to: {}", newRelicConfig.getBatchCount().get());
+        LOG.info("Flush delay set to: {}", newRelicConfig.getFlushDelay().get());
```

Comment:
Since this is infromative, could we be more verbose? like: 
LOG.info("Batch count set to: {} **messages**." ...
LOG.info("Flush delay set to: {} **seconds**."...
LOG.info("Parallelism set to: {} **thread**."...

(this comment it is completely useless, please feel free to throw me a 🌵 )

Reply:
Not sure either. I'll consider adding this to the toString, but we already provide documentation on the units of each configuration field, so we should be safe even if we don't include it here.

Reply:
Hey Noly, I opted to provide clear descriptions on the configuration fields in `NewRelicPipelineOptions`, and to make sure to specify that the units were mentioned there. I also switched to use a toString in `NewRelicConfig` for the logging, instead. Does this look good?